bfss3/bfss3.go

```diff
@@ -335,20 +340,26 @@ func normError(err error) error {
 		return nil
 	}
 
-	switch e := err.(type) {
-	case awserr.RequestFailure:
-		switch e.StatusCode() {
-		case http.StatusNotFound:
-			return bfs.ErrNotFound
-		}
-	case awserr.Error:
-		switch e.Code() {
-		case s3.ErrCodeNoSuchKey:
-			return bfs.ErrNotFound
-		case request.CanceledErrorCode:
-			return context.Canceled
-		}
-	}
+	// TODO: SDK v2 doesn't have many errors/error vars.
+	// Maybe most of them are now smth like:
+	// https://pkg.go.dev/github.com/aws/smithy-go/transport/http#example-ResponseError
+	// Check and convert properly.
+
+	// switch e := err.(type) {
+	// case awserr.RequestFailure:
+	// 	switch e.StatusCode() {
+	// 	case http.StatusNotFound:
+	// 		return bfs.ErrNotFound
+	// 	}
+	// case awserr.Error:
+	// 	switch e.Code() {
+	// 	case s3v1.ErrCodeNoSuchKey:
+	// 		return bfs.ErrNotFound
+	// 	case request.CanceledErrorCode:
+	// 		return context.Canceled
+	// 	}
+	// }
```

Comment:
TODO: important - handle/convert errors.

Want to play around with real S3 API for this to be sure.